File: mobile/concord/lib/api/models/send_request_response.dart
import 'enums.dart';
import 'json_utils.dart';

class SendRequestResponse {
  const SendRequestResponse({required this.requestId, required this.status});

  factory SendRequestResponse.fromJson(Map<String, dynamic> json) {
    return SendRequestResponse(
      requestId: json.field('RequestId') as String,
      status: FriendRequestStatus.fromWire(json.field('Status')),
    );
  }

  final String requestId;
  final FriendRequestStatus status;
}

File: backend/Concord/Concord.Application/Models/SendRequestResponse.cs
using System.Text.Json.Serialization;
using Concord.Application.Enums;

namespace Concord.Application.Models;

public class SendRequestResponse
{
    [JsonPropertyName("RequestId")]
    public Guid RequestId { get; set; }

    [JsonPropertyName("Status")]
    public FriendRequestStatus Status { get; set; }
}
